### ISSUE TITLE
Option "Print current document to HTML" overwrites markdown file contents

### ISSUE THREAD

**Issue by chanibal** [neilsustc/vscode-markdown#180]
When a markdown file has an extension other than `.md`, the option *Print current document to HTML* does not create a new HTML file, but replaces the current markdown file with HTML contents and saves.

How to recreate:
1. Open VSCode
2. Create a new file with an unused extensions, let's say `markdown.example`
3. Write some markdown into `markdown.example`
4. Command: *Change language mode*, select *markdown*
5. Command: *Markdown: Print current document to HTML*
Now, your `markdown.example` file has HTML contents instead of markdown - both in the VSCode editor window and on your hard drive.

Generating HTML from Markdown does not write anything meaningful in logs, but trying to undo it does (not sure if related, though):
```
mainThreadExtensionService.ts:43 [vscode.markdown-language-features]Illegal value for `line`
e.$onExtensionRuntimeError @ mainThreadExtensionService.ts:43
e._doInvokeHandler @ rpcProtocol.ts:265
e._invokeHandler @ rpcProtocol.ts:250
e._receiveRequest @ rpcProtocol.ts:196
e._receiveOneMessage @ rpcProtocol.ts:178
(anonymous) @ rpcProtocol.ts:111
(anonymous) @ rpcProtocol.ts:307
e.fire @ event.ts:140
(anonymous) @ ipc.net.ts:81
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560
```
```
mainThreadExtensionService.ts:44 Error: Illegal value for `line`
	at t._lineAt (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:460:691)
	at Object.lineAt (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:459:485)
	at getVisibleLine (c:\Program Files\Microsoft VS Code\resources\app\extensions\markdown-language-features\out\util\topmostLineMonitor.js:58:34)
	at MarkdownFileTopmostLineMonitor.vscode.window.onDidChangeTextEditorVisibleRanges.event (c:\Program Files\Microsoft VS Code\resources\app\extensions\markdown-language-features\out\util\topmostLineMonitor.js:18:30)
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:89:496)
	at e.$acceptEditorPropertiesChanged (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:690:540)
	at e._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:599:737)
	at e._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:599:455)
	at e._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:598:529)
	at e._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:598:298)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:597:226
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:600:324
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:88:119
	at e.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:89:496)
	at Socket.<anonymous> (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:145:1008)
	at emitOne (events.js:96:13)
	at Socket.emit (events.js:191:7)
	at readableAddChunk (_stream_readable.js:178:18)
	at Socket.Readable.push (_stream_readable.js:136:10)
	at Pipe.onread (net.js:560:20)
e.$onExtensionRuntimeError @ mainThreadExtensionService.ts:44
e._doInvokeHandler @ rpcProtocol.ts:265
e._invokeHandler @ rpcProtocol.ts:250
e._receiveRequest @ rpcProtocol.ts:196
e._receiveOneMessage @ rpcProtocol.ts:178
(anonymous) @ rpcProtocol.ts:111
(anonymous) @ rpcProtocol.ts:307
e.fire @ event.ts:140
(anonymous) @ ipc.net.ts:81
emitOne @ events.js:96
emit @ events.js:191
readableAddChunk @ _stream_readable.js:178
Readable.push @ _stream_readable.js:136
onread @ net.js:560
```

OS: Windows 10 1709  
Screenshot of the list of other extensions I have installed: https://i.imgur.com/0j4wMYG.png

**Comment by neilsustc** [comment]
Duplicate of #105. May I know why you need `.example` extension?

**Comment by chanibal** [comment]
Some software that I use uses another extension for content that is really just markdown text. 

Also, `.example` was just, well, an example extension. In reality there are a few cryptic file extensions that still are have markdown inside.

About #105, it's closed for over two months, but as far as I know, I have the most up to date version of the extension and vscode. Regression, or not yet fixed?

**Comment by neilsustc** [comment]
There was a targeted fix of `.MD` extension for #105.

Now you can download the latest fixed build [here](https://ci.appveyor.com/project/neilsustc/vscode-markdown/build/artifacts).

**Comment by neilsustc** [comment]
Fixed in v1.5.0
